repo/fsrepo/migrations/migrations.go

```diff
@@ -1,282 +1,208 @@
-package mfsr
+package migrations
 
 import (
-	"bufio"
-	"bytes"
+	"context"
 	"fmt"
-	"io"
 	"io/ioutil"
-	"net/http"
+	"log"
 	"os"
 	"os/exec"
-	"path/filepath"
+	"path"
 	"runtime"
-	"strconv"
 	"strings"
+	"sync"
 )
 
-var DistPath = "https://ipfs.io/ipfs/QmYRLRDKobvg1AXTGeK5Xk6ntWTsjGiHbyNKhWfz7koGpa"
+const (
+	// Migrations subdirectory in distribution. Empty for root (no subdir).
+	distMigsRoot = ""
+	distFSRM     = "fs-repo-migrations"
+)
 
-func init() {
-	if dist := os.Getenv("IPFS_DIST_PATH"); dist != "" {
-		DistPath = dist
+// RunMigration finds, downloads, and runs the individual migrations needed to
+// migrate the repo from its current version to the target version.
+func RunMigration(ctx context.Context, targetVer int, ipfsDir string) error {
+	ipfsDir, err := CheckIpfsDir(ipfsDir)
+	if err != nil {
+		return err
 	}
-}
-
-const migrations = "fs-repo-migrations"
-
-func migrationsBinName() string {
-	switch runtime.GOOS {
-	case "windows":
-		return migrations + ".exe"
-	default:
-		return migrations
+	fromVer, err := repoVersion(ipfsDir)
+	if err != nil {
+		return fmt.Errorf("could not get repo version: %s", err)
+	}
+	if fromVer == targetVer {
+		// repo already at target version number
+		return nil
 	}
-}
-
-func RunMigration(newv int) error {
-	migrateBin := migrationsBinName()
 
-	fmt.Println("  => Looking for suitable fs-repo-migrations binary.")
+	log.Print("Looking for suitable migration binaries.")
```

Comment:
Mostly, I wanted this diagnostic/progress output sent to stderr to prevent it from hindering any future conventional output that could serve as input to another program.  I also like having timestamps so that I can see how long individual items take to download, so I switched to using `log`.  Please let me know what you prefer.

This is probably part of a larger conversation about standardizing program output... do we user `log`, `stump`, `fmt`, etc, something else, and what output goes to what stream.